packages/terra-alert/src/terra-dev-site/test/alert/Alert.test.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import Button from 'terra-button';
+import Alert from '../../../Alert';
+
+class AlertActionButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      actionButtonClickCount: 0,
+      showAlert: false,
+    };
+    this.actionFunc = this.actionFunc.bind(this);
+    this.popAlert = this.popAlert.bind(this);
+  }
+
+  actionFunc() {
+    this.setState(prevState => ({ actionButtonClickCount: prevState.actionButtonClickCount + 1 }));
+  }
+
+  popAlert() {
+    this.setState(prevState => ({ showAlert: !prevState.showAlert }));
+  }
+
+  render() {
+    this.alert = (
+      <Alert id="actionAlert" type="warning" action={<Button text="Action" onClick={this.actionFunc} />}>
+    This is a warning. It is configured with a custom Action button. Action button has been
+    clicked
+        {' '}
+        <span id="actionButtonClickCount">{this.state.actionButtonClickCount}</span>
+        {' '}
+    times.
+      </Alert>
+    );
+    return (
+      <div>
+        {/* This Test has been added to test the functionality on JAWS and Voice Over */}
+        <Button text="Action" onClick={this.popAlert} />
+        {this.state.showAlert && this.alert }
+      </div>
```

Comment:
Should we add additional text for Voiceover/Jaws to read so we can see how the alert popping in is being handled? Does the screen reader move into the Alert, read the action and wait for the user to respond or move on? Does VoiceOver move back to the original position on the dom once the alert has been read?

Reply:
When the alert pops up, the VO stops reading the trigger button and starts reading the alert. After it has completed reading the alert it waits for user's response. If the alert is dismissed before VO has read the alert completely, then it reads the "Trigger Button" which was clicked to dismiss the alert.

Reply:
Is this consistent when using keyboard nav? I would expect the screen readers to be used with keyboard navigation instead of mouse

Reply:
Yes I tested it with keyboard nav specifically the VO navigation keys.It is consistent